control-plane/connect-inject/endpoints_controller.go

```diff
@@ -836,11 +839,17 @@ func (r *EndpointsController) processUpstreams(pod corev1.Pod) ([]api.Upstream,
 
 				// If Consul Namespaces are enabled, attempt to parse the
 				// upstream for a namespace.
-				if r.EnableConsulNamespaces {
-					pieces := strings.SplitN(parts[0], ".", 2)
-					serviceName = strings.TrimSpace(pieces[0])
-					if len(pieces) > 1 {
+				if r.EnableConsulNamespaces || r.EnableConsulPartitions {
+					pieces := strings.SplitN(parts[0], ".", 3)
```

Comment:
Is this the plan for Consul DNS too? `svc.ns.part`?

Reply:
yup! based on conversations with @freddygv Consul DNS will become `svc.ns.part`